### PR TITLE
Remove legacy local configuration for Button App position

### PR DESCRIPTION
**Pull request by nyonson**
## Description

I was about to add a *third* type of the wallet position so we can share it with the boost configurations, and that felt like a type too far. Proposing that we drop the legacy configuration support first instead to greatly simplify the logic. If earners reach out we can point them to the Earner App for the remote config settings.

### Other

* Bumped the nodejs version to get the test runner mocking fixes.
* Updated tests to key on `mount` method since the `_init` has been dropped.

### CODE REVIEW

File: packages/client-sdk/src/Mash.ts [hunk 1]
```diff
@@ -248,7 +242,7 @@ class Mash {
   /**
    * Mount the Button App iframe.
    */
-  private mount(position: MashWebAPI.WalletButtonPosition): Promise<null> {
+  async mount(position: MashWebAPI.WalletButtonPosition) {
```
Comment (nyonson) [inline packages/client-sdk/src/Mash.ts:245]: Need to expose this for the tests which doesn't feel great, but seems like a step in the right direction dropping `_init`.

File: packages/client-sdk/src/Mash.test.ts [hunk 2]
```diff
@@ -5,15 +5,6 @@ import Mash from "./Mash.js";
 import { MashEvent } from "./events.js";
 import { createDOM } from "./tests/dom.js";
 
-function mockMethod<T>(obj: unknown, methodName: string, fn: T) {
```
Comment (nyonson) [inline packages/client-sdk/src/Mash.test.ts:8]: @jstol just a heads up, new nodejs version works for this case

File: packages/client-sdk/src/Mash.ts [hunk 3]
```diff
@@ -143,23 +153,6 @@ class Mash {
     return curSpend + cleanCost <= maxSpend;
   }
 
-  private _init(settings?: MashSettings) {
-    // Just for backward compatibility with existing users who pass in settings.
-    if (settings) {
-      this.localConfig.earnerID = settings.id;
```
Comment (mdedys) [inline packages/client-sdk/src/Mash.ts:149]: Just curious, if we drop this. Will this break live implementations? I would be interested to test this out. Also interested if we can just drop `settings?: Mash Settings`  from line 167.

Comment (mdedys) [reply packages/client-sdk/src/Mash.ts:149]: Dropping the position is a "soft" break. Wallet will still show up but not in the right place. Removing earner id breaks all flows.

Comment (nyonson) [reply packages/client-sdk/src/Mash.ts:149]: Oh missed that. Let me wire up earner ID

Comment (mdedys) [reply packages/client-sdk/src/Mash.ts:149]: If we are dropping just position, then minor is cool with me.

**Review by mdedys — APPROVED** [review]

File: packages/client-sdk/src/Mash.test.ts [hunk 4]
```diff
@@ -41,18 +33,18 @@ test("Mash", async t => {
         autoHide: true,
       });
 
-      const _init = mockMethod(mash, "_init", async () => null);
+      const mount = mock.method(mash, "mount", async () => null);
```
Comment (cartonioh) [inline packages/client-sdk/src/Mash.test.ts:36]: curious why this now tests for `mount` vs the already public `init`

File: packages/client-sdk/src/Mash.ts (hunk 1 above)
Comment (cartonioh) [reply packages/client-sdk/src/Mash.ts:250]: why can't we test `init` which is part of this class's API?

**Review by mdedys — APPROVED** [review]

File: packages/client-sdk/src/Mash.ts (hunk 1 above)
Comment (cartonioh) [reply packages/client-sdk/src/Mash.ts:250]: just trying to understand the intention behind the change, previously we were testing `_init` which calls mount as part of its implementation, now we're actually testing the implementation detail

was the previous test wrong?

File: packages/client-sdk/src/Mash.test.ts (hunk 4 above)
Comment (nyonson) [reply packages/client-sdk/src/Mash.test.ts:36]: we are calling init in the test so wouldn't cover much heh. The test is checking when mount is called.

Comment (cartonioh) [reply packages/client-sdk/src/Mash.test.ts:36]: 👍 makes sense now, we were calling init in the test and then testing `_init`

but `_init` is now gone so we need to go deeper to test `mount`

File: packages/client-sdk/src/Mash.ts (hunk 1 above)
Comment (nyonson) [reply packages/client-sdk/src/Mash.ts:250]: We are trading the same evil. The `_init` method no longer does much so just combined it with `init`. What the test was checking for was *when* the mount is being called. Ideally, we could test this with some DI, but don't wanna do that refactor here.

**Review by cartonioh — APPROVED** [review]

**Pull request merged**